airflow/models/dag.py

```diff
@@ -355,7 +355,7 @@ def _create_orm_dagrun(
     "doc_md": str,
     "is_paused_upon_creation": bool,
     "render_template_as_native_obj": bool,
-    "tags": list,
+    "tags": abc.Collection,
```

Comment:
This should be a bit more preceise
```suggestion
    "tags": abc.Collection[str],
```

Reply:
That's what I have done before, but it isn't compatible with python 3.8. You can see it in [`PEP585`](https://peps.python.org/pep-0585/), it's only working on python 3.9+.

Maybe we should deprecate python 3.8?

Reply:
You can use `typing.Collection` instead.

Reply:
But [it would be deprecated in pytnon 3.9](https://docs.python.org/3.9/library/typing.html#typing.Collection). We would need to change it when we would drop support for python 3.8. I think sticking to `collections.abc` is a better option.

Reply:
But nice option.

Reply:
Since Airflow uses `from __future__ import annotation`, type hints become simply strings, so runtime deprecations do not matter as long as the imports are made in the `if TYPE_CHECKING` block.

What matters is _Mypy_ deprecating the support (it does not), not Python itself. Airflow already uses this import quite extensively.

Reply:
Okay, changed to `typing` instead of `collections.abc`. But would airflow move to `collections.abc` when it would deprecate `python 3.8`?

Reply:
Yes, might be in future. But `typing.Collection[str]` is also used in other places in the code. So this is not adding a new problem and need to be adjusted in the future anyway.